### DISCUSSION TITLE
Softening behavior

### DISCUSSION THREAD

**Discussion by SeyedSajjadMahdavi** [Anura3D/Anura3D_OpenSource#128] [Q&A] [answered]
Dear all,
How can we specify the peak and residual values related to the cohesion and friction angle or shape factor parameter of our material in Anura3d? 
What's the algorithm behind Strain smoothing Flag.
thx

**Comment by jacobosaa** [comment]
Hi seyed! 
i'm not an expert, but i hope this article that talks about your concern may help you!
[fern2016.pdf](https://github.com/Anura3D/Anura3D_OpenSource/files/12479991/fern2016.pdf)

**Comment by kmyost** [comment, marked as answer]
Hi Seyed!

You will need to implement a constitutive model that requires those parameters, for example, strain softening Mohr Coulomb. The current release of Anura3D does not come with a constitutive model that allows for softening so you would need to use an external soil model.

The strain smoothing feature is to reduce volumetric locking, it does not address softening behavior.

**Comment by SeyedSajjadMahdavi** [comment]
Hi kmyost,
Thank you for your reply and helpful response.
